src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -131,8 +130,16 @@ public static void dumpCoverage() throws IOException, MalformedObjectNameExcepti
     }
 
     @Before
-    public void cleanUpCache() throws Exception {
+    public void setUp() throws Exception {
+        super.setUp();
         clearCache();
+        resetDynamicSettings();
+    }
+
+    protected void resetDynamicSettings() throws IOException {
+        for (String setting : KNNSettings.dynamicCacheSettings.keySet()) {
+            updateClusterSettings(setting, null);
+        }
     }
```

Comment:
given that these setting(dynamicCacheSettings) are public static and tests run in parallel I am not sure if resetting the value of these settings will resolve the issue. As there can be tests which are accessing this value when we are reading these settings.

Reply:
I dont think rest tests run in parallel but I will double check.

Reply:
I believe its prevented here: https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/testclusters/StandaloneRestIntegTestTask.java#L108.

However, it seems that the cluster settings are already cleared after tests: https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L674. So this probably does not need to be called. Given that, I may close this PR. Seems we need to find another fix for #734

Reply:
Let's do a proper deep-dive and find out what is causing the issue.